src/formatDate.js

```diff
@@ -50,7 +50,49 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const currentSeparator = fromFormat[3];
+  const desiredSeparator = toFormat[3];
+  const currentFormat = fromFormat.slice(0, 3);
+  const desiredFormat = toFormat.slice(0, 3);
+  const dateArray = date.split(currentSeparator);
+  const day = dateArray[currentFormat.indexOf('DD')];
+  const month = dateArray[currentFormat.indexOf('MM')];
+  const result = [];
+  let year;
+
+  if (currentFormat.includes('YYYY')) {
+    year = dateArray[currentFormat.indexOf('YYYY')];
+  } else {
+    year = dateArray[currentFormat.indexOf('YY')];
+  }
+
+  if (currentFormat.includes('YY') && desiredFormat.includes('YYYY')) {
+    if (year < 30) {
+      year = '20' + year;
+    } else {
+      year = '19' + year;
+    }
+  }
+
+  if (currentFormat.includes('YYYY') && desiredFormat.includes('YY')) {
+    year = year.slice(2);
+  }
+
+  for (let i = 0; i < desiredFormat.length; i++) {
```

Comment:
Consider using 'switch' statement instead of multiple 'if' statements to improve code readability.